contract_variable_quantity/views/contract_line_views.xml

```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2019 Tecnativa - Pedro M. Baeza
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record id="contract_line_tree_view" model="ir.ui.view">
+        <field
+            name="name"
+        >contract.line tree view (in contract) - Add variable qty fields</field>
+        <field name="model">contract.line</field>
+        <field name="inherit_id" ref="contract.contract_line_tree_view" />
+        <field name="arch" type="xml">
+            <field name="quantity" position="before">
+                <field name="qty_type" />
+                <field
+                    name="qty_formula_id"
+                    invisible="[('qty_type', '!=', 'variable')]"
```

Comment:
column_invisible="qty_type != 'variable'"
Please check https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0

Reply:
The migration guide states:

```
On a tree view, if you have a field with invisible="1", you should change it to column_invisible="1", or now the column will be shown, but empty.
```

but this is making the visibility conditionally according some record values. That continues the same.

Reply:
I was about to add some comment in the migration guide, but this change comes from `attrs`, so nothing more to clarify, as it's totally different.